### PR TITLE
DD-1575 Dataverse CLI for running Dataverse integrity check

### PR DESCRIPTION
**Pull request by aliassheikh**
Fixes DD-1575 Dataverse CLI for running Dataverse integrity check

# Description of changes
added:
 - new dataset subcommnad: `validate-files`
 ```
[vagrant@archaeology ~]$ dataverse dataset --help
Usage: dataverse dataset [-hV] [-d=<delay>] targets [COMMAND]
Manage Dataverse datasets
      targets           The target(s) of the operation; this is either an ID a
                          file with a with a list of IDs, or - if the
                          subcommand supports it - a parameters file.
  -d, --delay=<delay>   Delay in milliseconds between requests to the server
                          (default: 1000).
  -h, --help            Show this help message and exit.
  -V, --version         Print version information and exit.
Commands:
  validate-files   Make sure that all files have correctly stored in object
                    storage.
``` 
```
[vagrant@archaeology ~]$ dataverse dataset validate-files --help
Usage: dataverse dataset validate-files [-hV]
 Make sure that all files have correctly stored in object storage.
  -h, --help      Show this help message and exit.
  -V, --version   Print version information and exit.
```

# How to test
**It needs admin access and can only be correctly tested in vagrant**:
```
$ vagrant ssh

[vagrant@archaeology ~]$ dataverse dataset doi:10.5072/DAR/AO7VWF validate-files
doi:10.5072/DAR/AO7VWF: OK. {"dataFiles": [
{"datafileId":9,"storageIdentifier":"file://190bc1dc8a2-7829c6eb9ca6","status":"valid"},
{"datafileId":10,"storageIdentifier":"file://190bc1dca10-4fa97bf643fc","status":"valid"},
{"datafileId":11,"storageIdentifier":"file://190bc1dcb3b-5c18980f5889","status":"valid"}
]
}
``` 
**In terminal, module path:**
```
..\dd-dataverse-cli $ start.sh dataset doi:10.5072/DAR/AO7VWF validate-files
doi:10.5072/DAR/AO7VWF: FAILED: Exception type = DataverseException, message = status: 503; message: { status:"error", message:"Endpoint available from localhost only. Please contact the dataverse administrator"} 
```

# Related PRs
(Add links)
*

# Notify
@DANS-KNAW/core-systems

### CODE REVIEW

File: src/main/java/nl/knaw/dans/dvcli/command/DatasetValidateFiles.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import java.util.List;
+
+@Command(name = "validate-files",
+         mixinStandardHelpOptions = true,
+         description = " Make sure that all files have correctly stored in object storage.")
+public class DatasetValidateFiles extends AbstractCmd {
+    @ParentCommand
+    private DatasetCmd datasetCmd;
+
+    protected List<Pair<String, IdParam>> getIds(DatasetCmd datasetCmd) throws IOException {
```
Comment (jo-pol) [inline src/main/java/nl/knaw/dans/dvcli/command/DatasetValidateFiles.java]: You can drop the argument and use the field above.

File: src/main/java/nl/knaw/dans/dvcli/command/DatasetValidateFiles.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        List<String> pids = new SingleIdOrIdsFile(datasetCmd.targets, SingleIdOrIdsFile.DEFAULT_TARGET_PLACEHOLDER).getPids().toList();
+        if (!pids.isEmpty())
```
Comment (jo-pol) [inline src/main/java/nl/knaw/dans/dvcli/command/DatasetValidateFiles.java]: I think you can drop this line and the second return.

File: src/main/java/nl/knaw/dans/dvcli/command/DatasetValidateFiles.java (hunk 1 above)
Comment (aliassheikh) [reply src/main/java/nl/knaw/dans/dvcli/command/DatasetValidateFiles.java]: Done.

File: src/main/java/nl/knaw/dans/dvcli/command/DatasetValidateFiles.java (hunk 2 above)
Comment (aliassheikh) [reply src/main/java/nl/knaw/dans/dvcli/command/DatasetValidateFiles.java]: Done.

**Review by jo-pol — APPROVED** [review]

**Pull request merged**